presto-native-execution/src/test/java/com/facebook/presto/nativeworker/NativeQueryRunnerUtils.java

```diff
@@ -96,12 +101,20 @@ public static void createAllIcebergTables(QueryRunner queryRunner)
     }
 
     public static void createLineitem(QueryRunner queryRunner)
+    {
+        createLineitem(queryRunner, true);
+    }
+
+    public static void createLineitem(QueryRunner queryRunner, boolean castDateToVarchar)
     {
         if (!queryRunner.tableExists(queryRunner.getDefaultSession(), "lineitem")) {
+            String shipDate = castDateToVarchar ? "cast(shipdate as varchar) as shipdate" : "shipdate";
```

Comment:
Is this needed by the CTE? If not, will you be able to move this change into a separate commit? Thanks.

Reply:
Yes, this is needed by CTE. 

I reused CTE Java engine tests for Native engine as well. The CTE tests use SQL with the right EXTRACT calls that depend on date columns being retained as DATE. All the native engine tests type-cast DATE columns to VARCHAR for both DWRF and Parquet in TPC-H schema.